src/users.spec.ts

```diff
@@ -235,38 +235,37 @@ describe('Users...', () => {
         expect(res.id).to.deep.equal(ctx.toJSON()['x-textile-thread'])
       })
     })
-    context('a developer with a user', () => {
+    context('a developer with a user', function () {
       const ctx: Context = new Context(addrApiurl, undefined)
       let dev: SignupReply.AsObject
-      const userContext = new Context(addrApiurl)
-      it('should sign-up, create an API key, and sign it for the requests', async () => {
+      let users: Users
+      it('should sign-up, create an API key, and a new user', async function () {
         // @note This should be done using the cli
         const { user } = await signUp(ctx, addrGatewayUrl, sessionSecret)
```

Comment:
So the new workflow becomes:
1. Via the CLI or eventual dashboard...
  a. Dev signs up etc
  b. They generate a user key
  c. They find some way to pass this key info to the app
2. Then in the app
  a. Create an identity (we use a random one by default)
  b. Create a new user from a key and an identity (with optional context if they are advanced)
  c. Create a thread for a bucket if they want (or even for a db)
  d. Create a bucket from a user, and you're good to go.

Reply:
All of the above are optional except the key bit. The signing etc is done under the hood. So in theory, this is a one-liner once they have a key and can serve it up to their app.

Reply:
Ok, with updates this is slightly more verbose, but cleaner thanks to some discussion with @sanderpick